mdsa/algorithm.py

```diff
@@ -0,0 +1,44 @@
+from abc import abstractmethod
+
+
+class Algorithm(object):
+    algorithms = {}
+
+    def __init__(self, algorithm_name):
+        self.name = algorithm_name
+
+    @abstractmethod
+    def run(self, distance_matrix, num_dimensions_out=10):
+        """
+        Performs approximation of Principal Coordinate Analysis (PcoA) using given algorithm.
+        Returns the principal coordinate results (not necessarily sorted).
+
+        Parameters
+        ----------
+        distance_matrix : np.array
```

Comment:
Awesome, didn't know this existed! Will port the code to use that instead – much more idiomatic indeed.